blog-task.rb

```diff
@@ -0,0 +1,111 @@
+# Copyright(C) 2024  Horimoto Yasuhiro <horimoto@clear-code.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+class BlogTask
+  include Rake::DSL
+
+  def initialize(package)
+    @package = package
+    @version = detect_version
+  end
+
+  def define
+    define_blog_task
+  end
+
+  private
+  def env_var(name, default=nil)
+    value = ENV[name] || default
+    raise "${#{name}} is missing" if value.nil?
+    value
+  end
+
+  def document_repository_path
+    raise NotImplementedError
+  end
+
+  def detect_version
+    raise NotImplementedError
+  end
+
+  def define_blog_task
+    namespace :document do
+      namespace :blog do
+        desc "Generate Blog from release note"
+        task :generate do
+          blog_file_name = "#{Time.now.strftime("%F")}-#{package}-#{version}.md"
+          File.open("#{document_repository_path}/ja/_posts/#{blog_file_name}", "w") do |blog_ja|
+            blog_ja.write(contents_ja(@package.capitalize, @version))
+          end
+
+          File.open("#{document_repository_path}/en/_posts/#{blog_file_name}", "w") do |blog_en|
+            blog_en.write(contents_en(@package.capitalize, @version))
+          end
+        end
+      end
+    end
+  end
+
+  def contents_ja(package, version)
+    major_version = version.split('.')[0];
```

Comment:
```suggestion
    major_version = version.split('.')[0]
```

あと文字列リテラルは基本的にダブルクォートにして。他のところではそうしているから。